src/coord/src/coord.rs

```diff
@@ -2287,7 +2269,35 @@ where
                 depends_on: sink.depends_on,
             }),
         };
-        match self.catalog_transact(vec![op], |_builder| Ok(())).await {
+
+        match self
```

Comment:
NIT: this is a huge expression for a `match`. I think it would be much clearer to first bind it to a variable and then match the variable